### PR TITLE
highlight symbols and their associated match to anything symbol like …

### PR DESCRIPTION
**Pull request by snuffysasa**
…a data offset when there is a likely mismatch

Tested on a few mips functions and ppc functions, not sure if others like ARM32 would throw any curve balls.

this is setup to affect the highlighting only, and not the scoring.

### CODE REVIEW

**Comment by simonlindholm** [comment]
why not affect scoring btw?

File: diff.py [hunk 1]
```diff
@@ -2020,13 +2018,11 @@ def process(dump: str, config: Config) -> List[Line]:
...
 
-        has_symbol = False
-
+        symbol = None
         while i < len(lines):
             reloc_row = lines[i]
             if re.search(arch.re_reloc, reloc_row):
-                original = processor.process_reloc(reloc_row, original)
-                has_symbol = True
+                original, symbol = processor.process_reloc(reloc_row, original)
```
[the remark this answers is not in the document]
Comment (snuffysasa) [reply diff.py:2025]: okay, can fix

File: diff.py [hunk 2]
```diff
@@ -2096,6 +2092,26 @@ def normalize_stack(row: str, arch: ArchSettings) -> str:
     return re.sub(arch.re_sprel, "addr(sp)", row)
 
 
+symbol_map: Dict[str, Any] = {}
```
[the remark this answers is not in the document]
Comment (snuffysasa) [reply diff.py:2095]: oh yea, meant to move this actually, will do

File: diff.py [hunk 3]
```diff
@@ -2189,15 +2205,18 @@ def diff_lines(
...
 
     old = old_line.scorable_line
     new = new_line.scorable_line
     if old == new:
-        return (0, 0)
+        return (0, 0, False)
 
     num_stack_penalties = 0
     num_regalloc_penalties = 0
+    symbol_mismatch_occured = False
```
[the remark this answers is not in the document]
Comment (snuffysasa) [reply diff.py:2219]: ok!

**Comment by snuffysasa** [comment]
> why not affect scoring btw?

well, I was thinking like, having the wrong symbols just means the data provided is incorrect, and not that the function is incorrectly decompiled.  Once you get it to the 100% point minus these mismatches... you would be able to easily change the data without affecting the function.

And also for the permuter, I don't think we want it to care about having incorrect external data provided, because that's something the user could clean up easily after the function is decompiled right.

I guess my mind could be changed on this though

File: diff.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonlindholm) [reply diff.py:2025]: still not fixed from what I can see (though None *is* better than "" so it's already an improvement)

File: diff.py [hunk 4]
```diff
@@ -2096,6 +2092,25 @@ def normalize_stack(row: str, arch: ArchSettings) -> str:
     return re.sub(arch.re_sprel, "addr(sp)", row)
 
 
+def check_for_symbol_mismatch(
+    old_line: Line, new_line: Line, symbol_map: Dict[str, Any]
```
Comment (simonlindholm) [inline diff.py:2096]: why Any and not `str`?

Comment (snuffysasa) [reply diff.py:2096]: oh we can do str.

Comment (snuffysasa) [reply diff.py]: updated

**Pull request merged**